qiskit/synthesis/linear/cnot_synth.py

```diff
@@ -53,25 +58,38 @@ def synth_cnot_count_full_pmh(
     """
     if not isinstance(state, (list, np.ndarray)):
         raise QiskitError(
-            f"state should be of type list or numpy.ndarray, but was of the type {type(state)}"
+            "state should be of type list or numpy.ndarray, "
+            "but was of the type {}".format(type(state))
         )
     state = np.array(state)
+    num_qubits = state.shape[0]
+
+    # Virtual Padding
```

Comment:
After discussing with @alexanderivrii and @ShellyGarion we came to the conclusion that the section size being larger than the number of qubits is an invalid input. It would be better to raise an error if `section_size > num_qubits`, as originally done in #12166. Could you do that and remove the virtual padding? 

We'll go forward with your PR then, as it still fixes the bad `section_size` values due to your fixes on L131 and L148.

Reply:
That's right, I did it